storage/src/main/java/io/delta/storage/S3SingleDriverLogStoreUtil.java

```diff
@@ -0,0 +1,47 @@
+package io.delta.storage;
+
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.fs.s3a.Listing;
+import org.apache.hadoop.fs.s3a.S3AFileStatus;
+import org.apache.hadoop.fs.s3a.S3AFileSystem;
+import org.apache.hadoop.fs.s3a.S3ListRequest;
+
+import java.io.IOException;
+import java.util.HashSet;
+
+import static org.apache.hadoop.fs.s3a.S3AUtils.ACCEPT_ALL;
+import static org.apache.hadoop.fs.s3a.S3AUtils.iteratorToStatuses;
+
+
+/**
+ * Static utility methods for the S3SingleDriverLogStore.
+ *
+ * Used to trick the class loader so we can use methods of org.apache.hadoop:hadoop-aws without needing to load this as
+ * a dependency for tests in core.
+ */
+class S3SingleDriverLogStoreUtil {
+    /**
+     * Uses the S3ListRequest.v2 interface with the startAfter parameter to only list files
+     * which are lexicographically greater than resolvedPath.
+     */
+    static FileStatus[] s3ListFrom(FileSystem fs, Path resolvedPath, Path parentPath) throws IOException {
+        S3AFileSystem s3afs = (S3AFileSystem) fs;
+        Listing listing = s3afs.getListing();
+        // List files lexicographically after resolvedPath inclusive within the same directory
+        RemoteIterator<S3AFileStatus> l = listing.createFileStatusListingIterator(resolvedPath,
+                S3ListRequest.v2(
+                        new ListObjectsV2Request()
+                                .withBucketName(s3afs.getBucket())
+                                .withMaxKeys(1000)
+                                .withPrefix(s3afs.pathToKey(parentPath))
+                                .withStartAfter(s3afs.pathToKey(resolvedPath))
```

Comment:
The difference between `listFrom` and `startAfter` is: `listFrom` is inclusive (including the path) but `startAfter` is exclusive (excluding the path).

Reply:
Okay, so this isn't specific just to `0000.json` it is for any `000N.json`.
We could also just use `io.delta.storage.internal.FileNameUtils` helper methods like `isDeltaFile`, `deltaVersion` to do some easy math: get the current delta version, subtract one.

Reply:
Do you have any pointers on how integration tests should be added for this? I've included an ignored tests which I've been using for local testing, but some setup is missing to be able to run this in CI.

Reply:
Hi @jkylling, here is one example of an integration test. https://github.com/delta-io/delta/blob/master/storage-s3-dynamodb/integration_tests/dynamodb_logstore.py.

However, your integration test for this code can instead go into `https://github.com/delta-io/delta/tree/master/storage` (into `/integration_tests`).

This integration test won't be run in the CI, but at least it lets us manually verify that the correct startAfter / listFrom logic is achieved.

Reply:
Would it be okay to adopt this test https://github.com/delta-io/delta/pull/1210/files#diff-0fc31cae0ffada2d2fdf10aa618310fcdf6c053876b5ac083b43a025f5084428R11 or should the tests be written in python?

Reply:
Adopting that test is perfectly fine! We just want this to run your actual new code, and not just revert to the previous implementation since it's on a local file system.

Reply:
Perfect! I'll go for that then

Reply:
I've added some integration tests for the tests for `S3LogStoreUtil.listFrom` method in `S3LogStoreUtilIntegrationTest`. You can run the tests with
```
S3_LOG_STORE_UTIL_TEST_BUCKET="some-bucket" S3_LOG_STORE_UTIL_TEST_KEY_PREFIX="some-key-prefix" python run-integration-tests.py --s3-log-store-util-only 
```